extensions/amp-fit-text/1.0/amp-fit-text.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FitText} from './fit-text';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {getLengthNumeral, isLayoutSizeDefined} from '../../../src/layout';
+import {isExperimentOn} from '../../../src/experiments';
+
+import {dict} from '../../../src/utils/object';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-fit-text';
+
+const getFontSizeAttrs = (element) =>
+  dict({
+    'maxFontSize': getLengthNumeral(element.getAttribute('max-font-size')),
+    'minFontSize': getLengthNumeral(element.getAttribute('min-font-size')),
+  });
+
+class AmpFitText extends PreactBaseElement {
+  /** @override */
+  init() {
+    const attributeOb = new MutationObserver(() => {
+      this.mutateProps(getFontSizeAttrs(this.element));
+    });
+    attributeOb.observe(this.element, {
+      attributeFilter: ['min-font-size', 'max-font-size'],
+      attributes: true,
+    });
+
+    // Force render to resize to new contents.
+    const childOb = new MutationObserver(() => {
+      this.mutateProps(dict({}));
+    });
+    childOb.observe(this.element, {
+      childList: true,
+      subtree: true,
+    });
+
+    const props = getFontSizeAttrs(this.element);
+    props['style'] = {
+      'width': this.element./* OK */ offsetWidth,
```

Comment:
So, here, all we need is `width: 100%, height: 100%`, right? That'd be very natural, won't require an extra measure, and will be fully auto-responsive.

Reply:
Resolving this thread to continue [here](https://github.com/ampproject/amphtml/pull/28169#discussion_r425297765)